src/java/org/apache/cassandra/io/util/FileUtils.java

```diff
@@ -350,13 +396,38 @@ public static String getRelativePath(String basePath, String path)
 
     public static void clean(ByteBuffer buffer)
     {
-        if (buffer == null)
+        if (buffer == null || !buffer.isDirect())
             return;
-        if (isCleanerAvailable && buffer.isDirect())
+
+        // TODO Once we can get rid of Java 8, it's simpler to call sun.misc.Unsafe.invokeCleaner(ByteBuffer),
+        // but need to take care of the attachment handling (i.e. whether 'buf' is a duplicate or slice) - that
+        // is different in sun.misc.Unsafe.invokeCleaner and this implementation.
+
+        try
         {
-            DirectBuffer db = (DirectBuffer) buffer;
-            if (db.cleaner() != null)
-                db.cleaner().clean();
+            if (buffer.isDirect())
```

Comment:
This check is redundant as you've already performed the `!buffer.isDirect()` earlier in this method (and bailed if it isn't)